File: typescript-api/src/data/services/create-todo.ts
import { Todo } from "../../domain/entities";
import { CreateTodoRepository } from "../repositories";

export class CreateTodoService {
  constructor(
    private readonly repo: CreateTodoRepository
  ) {}

  async create(userId: number, todo: Omit<Todo, 'id'>): Promise<void> {
    await this.repo.create(userId, todo)
  }
}

File: rust-api/src/data/services/create_todo.rs
use std::sync::Arc;

use crate::{data::repositories::CreateTodoRepository, domain::entities::Todo};

pub async fn create_todo_service(
    repo: Arc<impl CreateTodoRepository>,
    todo: Todo,
    user_id: i32
) -> Result<(), sqlx::Error> {
    repo.create(todo, user_id).await
}
